lib/src/linting/rules/no-optional-fields-within-responses.spec.ts

```diff
@@ -0,0 +1,169 @@
+import { HttpMethod } from "../../models/http";
+import {
+  ApiNode,
+  BodyNode,
+  EndpointNode,
+  ResponseNode
+} from "../../models/nodes";
+import {
+  DataType,
+  INT32,
+  NULL,
+  objectType,
+  STRING,
+  unionType
+} from "../../models/types";
+import { fakeLocatable } from "../../spec-helpers/fake-locatable";
+import { noOptionalFieldsWithinResponses } from "./no-optional-fields-within-responses";
+
+describe("rule: no optional fields within responses", () => {
```

Comment:
Both of the following can be interpreted as "optional":
```
{
  a?: string;
}

{
  a: string | null;
}
```

I prefer the term "omittable", which more clearly points to the former.